src/query/plan/operator.cpp

```diff
@@ -2880,6 +2904,11 @@ RemoveLabels::RemoveLabelsCursor::RemoveLabelsCursor(const RemoveLabels &self, u
 
 bool RemoveLabels::RemoveLabelsCursor::Pull(Frame &frame, ExecutionContext &context) {
   SCOPED_PROFILE_OP("RemoveLabels");
+  if (context.auth_checker &&
```

Comment:
Move this if and merge it with the one that checks update on vertex

Reply:
I missed an important part. This will result in two ifdefs and two licence checks

Reply:
But with this I will save a lot of checks for example let say we are setting/removing 2 labels and we have 1 milion nodes in database if we do it like this we will have only one check for both labels at start but if we do it how you propose we will have  1 milion checks for those same labels for each node that we pull. I would say that one additonal licence check and one additonal ifdef that is in compiling time is nothing compared to one milion check.

Reply:
Actually I think Boris is right. If you look at the interpreter.cpp file, you have this line of code

```
  for (; !n || i < n; ++i) {
    if (!pull_result()) {
      break;
    }

    if (!output_symbols.empty()) {
      stream_values();
    }
  }
```
if we have for example a query (1M nodes)
`MATCH (n) SET n:L2:L3 RETURN n;`
The operators would be
`Produce -> SetLabels -> ScanAll -> Once`

And the outer loop which I pasted above is responsible for doing it iteratively.
In the case of 1M nodes, when scanning the first node and going back to `SetLabels` operator, the query would fail either way at the check of `CREATE_DELETE` of `L2` and `L3`. The exception would propagate through the code mentioned above and it will no longer pull the other 999,999 nodes since the query is stopped.

Therefore it is good to group it since we are in fact not saving any significant performance.
Either way, benchmark results should tell us whether we missed something or not

Reply:
Okay but lets say for example you  have read permission on 500k node and you dont have create delete permission on any labels. In node cursor pull method you will have to check 500k nodes before you find that one node you can see and then you will check do you even have permission for creating deleting label while now how I did it you wont even start pulling nodes.